### ISSUE TITLE
Interaction with mitools

### ISSUE THREAD

**Issue by angelrodriguez2020** [amices/mice#260]
Dear fellows,
I have a set of imputed files created with MICE that I need to analyze with survey. When I use mitools to transfer them:

```
> library(mitools)
> imputedsurv<- imputationList(lapply(1:5, complete, x = imp))
```

This is what I get:

Error in UseMethod("complete_") : 
  no applicable method for 'complete_' applied to an object of class "c('integer', 'numeric')"
In addition: Warning message:
`complete_()` is deprecated as of tidyr 1.0.0.
Please use `complete()` instead.

Any idea why this is happening?

Thank you,

Ángel

**Comment by stefvanbuuren** [comment]
`mice` doesn't use the long-ago deprecated `tidy::complete_`. My suggestion is to direct your question to the `mitools` authors. Or live with the warning.

**Comment by angelrodriguez2020** [comment]
Dear Stef,

This is the answer I got from the author of mitools. Apparently, there is a typo in the syntax you suggest in: mice: Multivariate Imputation by Chained Equations in R. Journal of Statistical Software, December 2011, Volume 45, Issue 3.


"The code you were given 

imputedsurv<- imputationList(lapply(1:5, complete, x = imp))

is incorrect, because the data argument to mice::complete is actually called 'data'.  If you  do

imputedsurv<- imputationList(lapply(1:5, complete, data = imp))

it will work.   The reason for the mysterious calls to tidyr  is that mice imports the generic function complete() from tidyr.  If your call to complete() doesn't have a mids object for the data= argument (because you've called it 'x'), the complete.mids() method will not be called. Instead the tidyr::complete.default method will be called.

  -thomas

Thomas Lumley
Professor of Biostatistics
University of Auckland"

**Comment by stefvanbuuren** [comment]
Thanks for clearing this up.

1. I wasn't aware that the code you cited came from our mice 2011 JSS paper.
2. #212 changed `complete()` to evade interference with `tidyr::complete()`, and I had to rename the first argument. Back in 1999 when I wrote `complete()` I couldn't have dreamt that that would be needed some day. 
3. I cannot retrospectively change the JSS paper, so I think we need to live with it.
